README.md

```diff
@@ -105,6 +105,54 @@ aemHeadlessClient.runPersistedQuery('wknd/persist-query-name-with-variables', {
 })()    
 ```
 
+#### Pagination:
+```javascript
+(async () => {
+    const model = 'article'
+    const fields = `{
+        title
+        _path
+        authorFragment {
+          firstName
+          profilePicture {
+            ...on ImageRef {
+              _authorUrl
+            }
+          }
+        }
+      }`
+    
+    // Offset based: Loop all pages
+    const offsetQueryAll = await aemHeadlessClient.runPaginatedQuery(model, fields, { limit: 3 })
+    for await (let value of offsetQueryAll) {
+        console.log('offsetQueryAll', value)
+    }
+    // Offset based: Manually get next page
+    const offsetQuery = await aemHeadlessClient.runPaginatedQuery(model, fields, { limit: 3 })
+    while (true) {
+        const { done, value } = await offsetQuery.next();
+        if (done) break
+        console.log('offsetQuery', value)
+    }
+    // Cursor based: Loop all pages
+    const cursorQueryAll = await aemHeadlessClient.runPaginatedQuery(model, fields, { first: 4 })
+    for await (let value of cursorQueryAll) {
```

Comment:
Just for my understanding: This pattern uses the same approach under the hood than the one using `done`/`next()`.

Or, in other words: this pattern does not load everything before entering the loop, it uses lazy loading whenever there are not enough items (aka the next page) is read.

Otherwise it would be counterproductive to what we're trying to achieve with paging.

Reply:
It uses exactly the same approach as done/next, 

```const cursorQueryAll = await aemHeadlessClient.runPaginatedQuery```
is not making any requests.

Requests are in `for loop` for each page.